docs/arch/adr-03.md

```diff
@@ -0,0 +1,86 @@
+
+### Option 2: Azure File Service
+
+**Specs:**
+Target throughput for a single file share: up to 300 MiB/sec for certain regions, Up to 60 MiB/sec for all regions
+
+Pros: 
+- allows for specifying read-only or write-only permissions on folders within the share using a shared access signature (SAS): this would give us more control in how we want to organize/store the generated zip files
+- can cache Azure file shares on on-premises file servers by using Azure File Sync for quick access
+- File Services uses the SMB protocol, which is the same protocol used on file directories on Mac and Windows machines. Therefore a file share can be mapped onto a drive on your machine, which is not possible with a blob container
+- Greater potential throughput
+
+Cons:
+- Launched in Sept 2015, hasn't been around for as long as Blob Storage.
+- Smaller max size (100 TiB: 1 TiB = 2^40 bytes). We are not expecting to come close to this size, so this isn’t much of a concern
+
+## Decision
+
+While Azure File Service does offer the enticing option of granting folder specific permissions, we don't appear to have any need for the remainder of the additional functionality that comes with File Service, which makes me reluctant to want to use it. In addition, the number of articles and resources available on communicating with Blob Storage to set up file zipping is much greater than what's available for File Service. My initial analysis has not given me any reason to think that setting up user-specific containers would be problematic, but this is an open question worth thinking about for reviewers. Hypothetically, this user-specific container could be deleted after a short time window to avoid organizational clutter and unnecessary additional costs.
+
+Ultimately, my choice is to go with Blob Storage: the more basic, simple storage tool that gives us what we need and nothing more. That being said, I'd still like to keep the option of using Azure File Service on the table, in case it turns out that we *would* benefit from the additional functionality that it offers.
+
+Final questions:
+- Blob Storage doesn't have any concrete hierarchy beyond Storage Account/Blob Container - within a container, directories are virtual, demarcated by prefixes in the file name. Will this end up being problematic for us? Will it complicate file retrieval?
+- Would there be any benefit to caching files on on-premises file servers? If this sounds like something we'd like to employ, it would be worth reconsidering Azure File Service
+- Is there any reason why we might not want to create a new container every time a user wants to download a set of files? It doesn't seem organizationally ideal to me since this would mean that the Tove production storage account would end up cluttered with user-specific folders on the top level of its blob storage, but if we delete them after giving the user a window of time to download, it probably would not get out of hand.
```

Comment:
I think we'd create 1 top level container for the tove application data and store the required file/blob data there.

As our requests come in for a data export, we'd create / enumerate all the files (using a path schema) we need for the export, download them create a temporary zip together, upload that zip to it's own path and maybe email the requester a time limited SAS blob URL for that zip file or provide a time limited SAS url on application access via a link.

If you want to get tricky you could look at creating some zip file path based on the incoming request and tove transcription data to use existing built exports if they exist (maybe a hash of the params to create a unique file key). This would be a form of caching so may not work as new data comes in and it might be easier to reconstruct the export each time and replace the existing blob with the new one.

Reply:
>I think we'd create 1 top level container for the tove application data and store the required file/blob data there.

Definitely, agreed. The only reason I had been thinking about creating a new container for each user was because of the permissions restrictions (which turned out to be wrong). With the ability to grant blob-specific permissions, we can definitely store all generated files in the same container.

Based on discussing this with @zwolf before the holidays, we might not have access to the user's email (though I'm not sure on the specifics of why this is and what would need to be done in order to get access to it - perhaps this a current limitation of the Panoptes API?). I think this discussion would be worth shelving until we have an idea of how quickly we're able to download/zip all files for a given project/workflow/etc. In an ideal scenario, we would be able to trigger a direct download of the zip file.